src/components/common/Extensions/ExtensionSetup/ExtensionSetup.tsx

```diff
@@ -0,0 +1,197 @@
+  <span className={styles.descriptionExample}>{chunks}</span>
+);
+
+interface Props {
+  extensionData: InstalledExtensionData;
+}
+
+const ExtensionSetup = ({
+  extensionData,
+  extensionData: {
+    isInitialized,
+    isDeprecated,
+    extensionId,
+    initializationParams,
+  },
+}: Props) => {
+  const navigate = useNavigate();
+  const { colony } = useColonyContext();
+
+  const transform = pipe(
+    mapPayload((payload) => {
+      if (extensionId === Extension.VotingReputation) {
+        return initializationParams?.reduce(
+          (formattedPayload, { paramName }) => {
+            if (paramName.endsWith('Period')) {
+              return {
+                ...formattedPayload,
+                [paramName]: new Decimal(payload[paramName])
+                  .mul(3600)
+                  .toFixed(0, Decimal.ROUND_HALF_UP),
+              };
+            }
+            return {
+              ...formattedPayload,
+              [paramName]: new Decimal(payload[paramName])
+                .mul(new Decimal(10).pow(16))
+                .toString(),
+            };
+          },
+          {},
+        );
+      }
+
+      return payload;
+    }),
+    mergePayload({ colonyAddress: colony?.colonyAddress, extensionData }),
+  );
+
+  const handleFormSuccess = useCallback(() => {
+    navigate(`/colony/${colony?.name}/extensions/${extensionId}`);
+  }, [colony?.name, extensionId, navigate]);
+
+  const initialValues = useMemo(() => {
+    if (!initializationParams) {
+      return {};
+    }
+    return createExtensionSetupInitialValues(initializationParams);
+  }, [initializationParams]);
+
+  if (!colony) {
+    return null;
+  }
+
+  if (isInitialized || isDeprecated || !initializationParams) {
+    return <Navigate to={`/colony/${colony.name}/extensions/${extensionId}`} />;
+  }
+
+  const renderParams = (
+    // using any since this will be refactored into react-hook-form anyway
+    formikProps: FormikProps<any>,
+  ) =>
+    initializationParams.map(
+      ({ paramName, type, title, description, complementaryLabel }) => (
+        <div key={paramName}>
+          {type === ExtensionParamType.Input && (
+            <div className={styles.input}>
+              <Input
+                appearance={{ size: 'medium', theme: 'minimal' }}
+                label={title}
+                name={paramName}
+                disabled={formikProps.isSubmitting}
+              />
+              <p className={styles.inputsDescription}>
+                <FormattedMessage
+                  {...description}
+                  values={{
+                    span: DescriptionChunks,
+                  }}
+                />
+              </p>
+              {complementaryLabel && (
+                <span className={styles.complementaryLabel}>
+                  <FormattedMessage {...MSG[complementaryLabel]} />
+                </span>
+              )}
+            </div>
+          )}
+        </div>
```

Comment:
This looks like a good opportunity to extract into a separate component for the InitializationParam, what do you think?

Reply:
Agree, thanks!